contracts/extensions/BatchTradeExtension.sol

```diff
@@ -0,0 +1,201 @@
+    }
+
+    /**
+     * ONLY OWNER: Initializes BatchTradeExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external onlyOwnerAndValidManager(_delegatedManager) {
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        _initializeExtension(setToken, _delegatedManager);
+
+        emit BatchTradeExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY OWNER: Initializes TradeExtension to the DelegatedManager and TradeModule to the SetToken
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeModuleAndExtension(IDelegatedManager _delegatedManager) external onlyOwnerAndValidManager(_delegatedManager){
+        require(_delegatedManager.isPendingExtension(address(this)), "Extension must be pending");
+
+        ISetToken setToken = _delegatedManager.setToken();
+
+        _initializeExtension(setToken, _delegatedManager);
+        _initializeModule(setToken, _delegatedManager);
+
+        emit BatchTradeExtensionInitialized(address(setToken), address(_delegatedManager));
+    }
+
+    /**
+     * ONLY MANAGER: Remove an existing SetToken and DelegatedManager tracked by the BatchTradeExtension
+     */
+    function removeExtension() external override {
+        IDelegatedManager delegatedManager = IDelegatedManager(msg.sender);
+        ISetToken setToken = delegatedManager.setToken();
+
+        _removeExtension(setToken, delegatedManager);
+    }
+
+    /**
+     * ONLY OPERATOR: Executes a trade on a supported DEX.
+     * @dev Although the SetToken units are passed in for the send and receive quantities, the total quantity
+     * sent and received is the quantity of SetToken units multiplied by the SetToken totalSupply.
+     *
+     * @param _setToken             Instance of the SetToken to trade
+     * @param _trades               Struct of information for individual trades
+     */
+    function trade(
+        ISetToken _setToken,
+        TradeInfo[] memory _trades
+    )
+        external
+        onlyOperator(_setToken)
+        onlyAllowedAssets(_setToken, _trades)
+    {
+        for(uint256 i = 0; i < _trades.length; i++) {
+            bytes memory callData = abi.encodeWithSignature(
+                "trade(address,string,address,uint256,address,uint256,bytes)",
+                _setToken,
+                _trades[i].exchangeName,
+                _trades[i].sendToken,
+                _trades[i].sendQuantity,
+                _trades[i].receiveToken,
+                _trades[i].minReceiveQuantity,
+                _trades[i].data
+            );
+
+            try _manager(_setToken).interactManager(address(tradeModule), callData) {}
```

Comment:
Note: here we don't use `_invokeManager` because `try` can only be used with external function calls and contract creation calls